tests/integration/test_check_downloaded_output.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import subprocess
+from os import path
+import tarfile
+from pathlib import Path
+
+import utils
+
+
+def test_check_downloaded_output(test_env):
+    """
+    Check that the bundle has all the necessities.
+
+    Process:
+    * Send new request to cachito api
+    * Send request to download appropriate bundle from cachito
+
+    Checks:
+    * Check that response code is 200
+    * Check that state is "complete"
+    * Check the downloaded data are in gzip format and valid
+    * Check that dir deps/gomod/… contains cached dependencies
+    * Check that dir app/ contains application source code
+    * Check that the same full path filename is not duplicated
+    """
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    response = client.create_new_request(
+        payload={
+            "repo": test_env["download_output"]["repo"],
+            "ref": test_env["download_output"]["ref"],
+            "pkg_managers": test_env["download_output"]["pkg_managers"],
+        },
+    )
+
+    assert response.status == 201
+    response = client.wait_for_complete_request(response)
+    assert response.data["state"] == "complete"
+
+    resp = client.download_bundle(response.id)
+    assert resp.status == 200
+
+    file_name = f"download_{str(response.id)}"
+    file_name_tar = f"{file_name}.tar.gz"
+    with open(file_name_tar, "wb") as file:
+        file.write(resp.data)
+
+    process = subprocess.run(["gzip", "-t", file_name_tar])
+    assert process.returncode == 0
+
+    with tarfile.open(file_name_tar, "r:gz") as tar:
+        tar.extractall(file_name)
+
+    dependencies_path = "deps/gomod/pkg/mod/cache/download/"
+    names = [i["name"] for i in response.data["dependencies"]]
+    for dependency in names:
+        package_name = utils.escape_path_go(dependency)
+        dependency_path = path.join(file_name, dependencies_path, package_name)
+        assert path.exists(dependency_path)
+
+    with open(f"{file_name}/app/go.mod", "r") as file:
+        module_name = list()
+        for line in file:
+            if "module" in line:
```

Comment:
How about the following to be a little more precise?
```suggestion
            if line.startswith("module "):
```

I'm just concerned that some other line would have the text `module` in it such as a comment.

Reply:
that's better :)